NickvisionTubeConverter.Shared/Models/Configuration.cs

```diff
@@ -98,6 +98,10 @@ public Configuration()
         {
             Directory.CreateDirectory(ConfigDir);
         }
+        if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("SNAP_USER_COMMON")))
+        {
+        PreviousSaveFolder = Environment.GetEnvironmentVariable("SNAP_USER_COMMON");
+        }
         Theme = Theme.System;
         PreviousSaveFolder = Environment.GetFolderPath(Environment.SpecialFolder.MyVideos);
```

Comment:
Move your addition below, currently the path will be replaced with `Environment.SpecialFolder.MyVideos` anyway